islandora_web_archive.module

```diff
@@ -181,3 +181,39 @@ function islandora_web_archive_islandora_sp_web_archive_islandora_derivative() {
     ),
   );
 }
+
+/**
+ * Implements hook_cmodel_pid_dsid_islandora_datastream_alter().
+ */
+function islandora_web_archive_islandora_sp_web_archive_png_islandora_datastream_alter(AbstractObject $object, AbstractDatastream $datastream, array &$context) {
+  // When the PNG datastream is updated update the screenshot datastream with
+  // the updated PNG file.
+  if ($context['action'] == 'ingest') {
+    if (!isset($object['SCREENSHOT'])) {
+      module_load_include('inc', 'islandora_web_archive', 'includes/utilities');
+      islandora_web_archive_clone_datastream($object, $object['PNG'], 'SCREENSHOT');
```

Comment:
Couldn't a case be possible where a SCREENSHOT isn't present an a PNG is not present?